### PR TITLE
dvclive: Add plots

### PR DESCRIPTION
**Pull request by daavoo**
P.R. for https://github.com/iterative/dvclive/pull/189

### CODE REVIEW

**Comment by jorgeorpinel** [comment]
Interesting! Please request a review from maybe @dberenbaum or perhaps @casperdcl when ready. Thanks

File: content/docs/dvclive/api-reference/sklearn.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
+# Scikit-learn utilities
+
+See [detailed usage examples](/doc/dvclive/ml-frameworks/sklearn).
+
+## sklearn.log_confusion_matrix
+
+Generates a [_Confusion Matrix_](https://en.wikipedia.org/wiki/Confusion_matrix)
+plot (usable by `dvc plots`) .
```
Comment (jorgeorpinel) [inline content/docs/dvclive/api-reference/sklearn.md:8]: 💅 If you're already using `_` a lot in the DVCLive ref then it's OK but probably no need for italics in linked text that already stands out due to underscore and blue color. If anything it should be a CSS style

Comment (jorgeorpinel) [inline content/docs/dvclive/api-reference/sklearn.md:8]: Usable by `dvc plots` how? Rhetorical: please clarify in the doc :)

Comment (jorgeorpinel) [reply content/docs/dvclive/api-reference/sklearn.md:8]: > Usable by dvc plots how?

Nvmd I see it's explained in the description. I'd just maybe put it like this instead:

```suggestion
Generates a [confusion matrix](https://en.wikipedia.org/wiki/Confusion_matrix)
(which can be visualized as `dvc plots`) .
```

File: content/docs/dvclive/api-reference/sklearn.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+y_true = [1, 1, 2, 2]
+y_pred = [2, 1, 1, 2]
+
+log_confusion_matrix(y_true, y_score, "cm.json")
+```
+
+### Description
+
+Stores the predictions and labels in the format expected by the DVC
+[confusion matrix template](/doc/command-reference/plots#example-confusion-matrix).
```
Comment (jorgeorpinel) [inline content/docs/dvclive/api-reference/sklearn.md:28]:
```suggestion
Stores the predictions and labels in the format expected by the confusion
matrix
[template](/doc/command-reference/plots#plot-templates-data-series-only)
of `dvc plots`.
```

File: content/docs/dvclive/api-reference/sklearn.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+### Parameters
+
+- `y_true` - Ground truth (correct) target values.
+
+- `y_score` - Estimated targets as returned by a classifier..
+
+- `output_file` - The name of the file where the `.json` plot will be saved.
```
Comment (jorgeorpinel) [inline content/docs/dvclive/api-reference/sklearn.md:36]: 💅 The capitalization can work either way but in DVC docs we use lower case in these (up to you). The periods are not needed here though (incomplete sentences).
```suggestion
- `y_true` - ground truth (correct) target values

- `y_score` - estimated targets as returned by a classifier

- `output_file` - name of the file where the JSON plot will be saved
```

Comment (jorgeorpinel) [inline content/docs/dvclive/api-reference/sklearn.md:36]: First mention that the output is JSON. Should be stated earlier, I think.

File: content/docs/dvclive/ml-frameworks/sklearn.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+DVCLive allows you to easily add experiment tracking capabilities to your
+Scikit-learn projects.
+
+## About Scikit-learn
+
+[Scikit-learn](https://scikit-learn.org/) is an open source machine learning
+library that supports supervised and unsupervised learning. It also provides
+various tools for model fitting, data preprocessing, model selection, model
+evaluation, and many other utilities.
```
Comment (jorgeorpinel) [inline content/docs/dvclive/ml-frameworks/sklearn.md:11]: Why do we want these sections that document 3rd party tools?

File: content/docs/dvclive/api-reference/sklearn.md (hunk 2 above)
Comment (jorgeorpinel) [reply content/docs/dvclive/api-reference/sklearn.md:28]: Or maybe use your description from the other page. Seems more straightforward:

> set of... utilities that allow you to generate `dvc plots` from the labels and predictions...

File: content/docs/dvclive/ml-frameworks/sklearn.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+## Usage
+
+To start using DVCLive you just need to add few modifications to your training
```
Comment (jorgeorpinel) [inline content/docs/dvclive/ml-frameworks/sklearn.md:15]: Avoid minimizing importance/complexity -- can be perceived as condescending.
```suggestion
To start using DVCLive, add a few modifications to your training
```

File: content/docs/dvclive/ml-frameworks/sklearn.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+from dvclive.sklearn import (
+    log_confusion_matrix,
+    log_precision_recall,
+    log_roc,
+)
+
+roc = log_roc(y_test, y_score, "roc.json")
+prc = log_precision_recall(y_test, y_score, "prc.json")
+cm = log_confusion_matrix(y_test, y_pred, "cm.json")
+
```
Comment (jorgeorpinel) [inline content/docs/dvclive/ml-frameworks/sklearn.md:50]:
```suggestion
```

File: content/docs/dvclive/ml-frameworks/sklearn.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+library that supports supervised and unsupervised learning. It also provides
+various tools for model fitting, data preprocessing, model selection, model
+evaluation, and many other utilities.
+
+## Usage
+
+To start using DVCLive you just need to add few modifications to your training
+code in **any** Scikit-learn project.
+
+To ilustrate with some code, given the following example:
```
Comment (jorgeorpinel) [inline content/docs/dvclive/ml-frameworks/sklearn.md:18]:
```suggestion
To illustrate with some code, let's consider the following example:
```

File: content/docs/dvclive/ml-frameworks/sklearn.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+clf = RandomForestClassifier(random_state=0)
+clf.fit(X_train, y_train)
+
+y_pred = clf.predict(X_test)
+y_score = clf.predict_proba(X_test)[:, 1]
+```
+
+DVCLive provides a set of
+[Scikit-learn utilities](/doc/dvclive/api-reference/sklearn) that allow you to
+generate `dvc plots` from the labels and predictions.
```
Comment (jorgeorpinel) [inline content/docs/dvclive/ml-frameworks/sklearn.md:36]:
```suggestion
generate `dvc plots` from your labels and predictions.
```

File: content/docs/dvclive/ml-frameworks/sklearn.md [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+
+```python
+...
+
```
Comment (jorgeorpinel) [inline content/docs/dvclive/ml-frameworks/sklearn.md:40]: Prob not needed since we later ask the user to stitch the snippets together
```suggestion
```

File: content/docs/dvclive/ml-frameworks/sklearn.md (hunk 6 above)
Comment (jorgeorpinel) [inline content/docs/dvclive/ml-frameworks/sklearn.md:50]:
```suggestion
```

File: content/docs/dvclive/ml-frameworks/sklearn.md [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+roc = log_roc(y_test, y_score, "roc.json")
+prc = log_precision_recall(y_test, y_score, "prc.json")
+cm = log_confusion_matrix(y_test, y_pred, "cm.json")
+
+```
+
+Joining the above snippets into a single script named `train.py`, we can
+integrate with DVC without further modifications to the script.
+
+We just need to define a `dvc.yaml` stage as follows:
```
Comment (jorgeorpinel) [inline content/docs/dvclive/ml-frameworks/sklearn.md:56]:
```suggestion
We can integrate with DVC by joining the above snippets into a single script
(e.g. named `train.py`). We also need to write a `dvc.yaml` file as follows:
```

File: content/docs/dvclive/api-reference/sklearn.md [hunk 11]
```diff
@@ -0,0 +1,214 @@
+# Scikit-learn utilities
+
+See [detailed usage examples](/doc/dvclive/ml-frameworks/sklearn).
```
Comment (jorgeorpinel) [inline content/docs/dvclive/api-reference/sklearn.md:3]: Maybe
```suggestion
> See also a [detailed usage examples](/doc/dvclive/ml-frameworks/sklearn).
```

**Review by jorgeorpinel — APPROVED** [review]
Sorry for the delay.

File: content/docs/dvclive/ml-frameworks/index.md [hunk 12]
```diff
@@ -13,6 +13,7 @@ We currently support the following _ML Frameworks_:
 - [MMCV](/docs/dvclive/ml-frameworks/mmcv)
 - [PyTorch](/docs/dvclive/ml-frameworks/pytorch)
 - [PyTorch Lightning](/docs/dvclive/ml-frameworks/pytorch-lightning)
+- [Scikit-Learn](/docs/dvclive/ml-frameworks/sklearn)
```
Comment (casperdcl) [inline content/docs/dvclive/ml-frameworks/index.md:16]: for consistency with the other mentions

```suggestion
- [Scikit-learn](/docs/dvclive/ml-frameworks/sklearn)
```

File: content/docs/dvclive/api-reference/plots.md [hunk 13]
```diff
@@ -0,0 +1,214 @@
+# Plotting utilities
+
+Generates
+[Scikit-learn visualizations](https://scikit-learn.org/stable/visualizations.html)
```
Comment (casperdcl) [inline content/docs/dvclive/api-reference/plots.md:4]:
```suggestion
[scikit-learn visualizations](https://scikit-learn.org/stable/visualizations.html)
```

File: content/docs/dvclive/ml-frameworks/sklearn.md [hunk 14]
```diff
@@ -0,0 +1,90 @@
+# Scikit-learn
+
+DVCLive allows you to add experiment tracking capabilities to your
+[Scikit-learn](https://scikit-learn.org/) projects.
+
+## Usage
+
+To start using DVCLive, add a few modifications to your training code in **any**
+[Scikit-learn](https://scikit-learn.org/) project.
```
Comment (casperdcl) [inline content/docs/dvclive/ml-frameworks/sklearn.md:9]:
```suggestion
[scikit-learn](https://scikit-learn.org/) projects.

## Usage

To start using DVCLive, add a few modifications to your training code in **any**
[scikit-learn](https://scikit-learn.org/) project.
```

**Review by casperdcl — COMMENTED** [review]
lgtm; just minor naming/capitalisation

File: content/docs/dvclive/api-reference/plots.md (hunk 13 above)
Comment (jorgeorpinel) [reply content/docs/dvclive/api-reference/plots.md:4]: I committed most of these but this one (and another similar one) I'd also capitalize since Scikit-learn is a proper noun. Or we could go with the brand's preferred lower-case `scikit-learn` everywhere (like we do with "GitHub" in DVC docs at least). Ultimately as long as it's consistent either way works so up to you @daavoo
